packages/aws-cdk/lib/cli.ts

```diff
@@ -458,14 +467,21 @@ async function initCommandLine() {
       case 'synthesize':
       case 'synth':
         if (args.exclusively) {
-          return cli.synth(args.STACKS, args.exclusively, args.quiet, args.validation);
+          return cli.synth(args.STACKS, args.exclusively, args.quiet, args.validation, argv.json);
         } else {
-          return cli.synth(args.STACKS, true, args.quiet, args.validation);
+          return cli.synth(args.STACKS, true, args.quiet, args.validation, argv.json);
         }
 
+      case 'notices':
+        // This is a valid command, but we're postponing its execution
```

Comment:
I would have put the command here, but I guess you're not doing that on purpose.

Reply:
Yes, if we put the command here, it will show the notices again at the end, unless we have some way of signaling that notices were already shown and there is no need to show it again. I don't like either approach, but this one seems a bit cleaner.